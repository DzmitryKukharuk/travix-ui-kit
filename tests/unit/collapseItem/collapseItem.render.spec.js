import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import CollapseItem from '../../../components/collapse/collapseItem';

describe('Collapse: render', () => {
  it('should render collapse item component', () => {
    const component = shallow(
      <CollapseItem
        id="c1"
        name="test-collapse-item"
        onChange={jest.fn()}
        title="Collapse Item 1"
      >
        <div>Collapse Item content</div>
      </CollapseItem>
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should render expanded collapse item component', () => {
    const component = shallow(
      <CollapseItem
        expanded
        id="c1"
        name="test-collapse-item"
        onChange={jest.fn()}
        title="Collapse Item 1"
      >
        <div>Collapse Item content</div>
      </CollapseItem>
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should render collapse item component with correct type for accordion mode', () => {
    const component = shallow(
      <CollapseItem
        accordion
        id="c1"
        name="test-collapse-item"
        onChange={jest.fn()}
        title="Collapse Item 1"
      >
        <div>Collapse Item content</div>
      </CollapseItem>
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should call onChange after click on title', () => {
    const e = { target: { checked: true } };
    const onChange = jest.fn();
    const component = shallow(
      <CollapseItem
        id="c1"
        name="test-collapse-item"
        onChange={onChange}
        title="Collapse Item 1"
      >
        <div>Collapse Item content</div>
      </CollapseItem>
    );
    component.find('.ui-collapse-item__input').simulate('change', e);
    expect(onChange).toBeCalledWith(e);
  });
});
