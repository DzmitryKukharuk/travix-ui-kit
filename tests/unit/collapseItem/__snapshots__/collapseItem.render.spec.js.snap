// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse: render should render collapse item component 1`] = `
<div
  className="ui-collapse-item"
>
  <input
    className="ui-collapse-item__input"
    id="c1"
    name="test-collapse-item"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="ui-collapse-item__label"
    htmlFor="c1"
  >
    Collapse Item 1
  </label>
  <div
    className="ui-collapse-item__content"
  >
    <div>
      Collapse Item content
    </div>
  </div>
</div>
`;

exports[`Collapse: render should render collapse item component with correct type for accordion mode 1`] = `
<div
  className="ui-collapse-item"
>
  <input
    className="ui-collapse-item__input"
    id="c1"
    name="test-collapse-item"
    onChange={[Function]}
    type="radio"
  />
  <label
    className="ui-collapse-item__label"
    htmlFor="c1"
  >
    Collapse Item 1
  </label>
  <div
    className="ui-collapse-item__content"
  >
    <div>
      Collapse Item content
    </div>
  </div>
</div>
`;

exports[`Collapse: render should render expanded collapse item component 1`] = `
<div
  className="ui-collapse-item"
>
  <input
    className="ui-collapse-item__input"
    defaultChecked={true}
    id="c1"
    name="test-collapse-item"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="ui-collapse-item__label"
    htmlFor="c1"
  >
    Collapse Item 1
  </label>
  <div
    className="ui-collapse-item__content"
  >
    <div>
      Collapse Item content
    </div>
  </div>
</div>
`;
